### ISSUE TITLE
Fix Request: False positive in the NotProperlySynchronizingOnThisWhileUsingGuardedBy rule

### ISSUE THREAD

**Issue by Syuziko** [jborgers/PMD-jPinpoint-rules#576]
The variable of the violation does not have the GuardedBy annotation, yet an inner static class with a variable with the same name does have the GuardeBy annotation. So the rule gets confused.

**Comment by jborgers** [comment]
It is a false positive, the rule does not differentiate between inner and outer class, both matches are accepted.
example: 
```
import net.jcip.annotations.GuardedBy;

class Issue576 {
    @GuardedBy("this")
    private String txt1;
    private String txt2;

    public void modifyTextSafe() {
        synchronized(this) {
            txt1 += "-appended";
            txt2 += "-appended";
        }
    }

    public boolean modifyTextUnSafe() {
      txt1 += "-appended"; // bad
      txt2 += "-appended"; // <-- false positive
    }

    static class InnerIssue {
        private String txt1;
        @GuardedBy("this")    
        private String txt2;

        public void modifyTextSafe() {
          synchronized(this) {
            txt1 += "-appended";
            txt2 += "-appended";
          }
      }

      public boolean modifyTextUnSafe() {
        txt1 += "-appended"; // <-- false positive
        txt2 += "-appended"; // bad
      }
    }
}

```
